packages/client/src/clients/in-app-messages/message-client.ts

```diff
@@ -8,16 +8,32 @@ import { InAppChannelClient } from "./channel-client";
 import {
   FetchInAppMessagesOptions,
   InAppMessage,
+  InAppMessageEngagementStatus,
   InAppMessageResponse,
   InAppMessageStoreState,
   InAppMessagesClientOptions,
 } from "./types";
 
+const engagementFilterRules: Record<
+  InAppMessageEngagementStatus,
+  (msg: InAppMessage) => boolean
+> = {
+  read: (msg) => Boolean(msg.read_at),
+  unread: (msg) => !msg.read_at,
+  seen: (msg) => Boolean(msg.seen_at),
+  unseen: (msg) => !msg.seen_at,
```

Comment:
Yeah, with the current behavior it could render through all current messages but I don't think it'd crash the browser (like say a component bug causing infinite renders). 

Let's discuss as a team what guards we should, if any, to marking as seen. It could be as simple as adding a delay, ensuring the component is actually in the viewport, etc.

Reply:
Thinking on this a bit more, I guess the underlying issue isn't limited to the auto markAsSeen() call specifically? I'm thinking about a different situation where user interacts with a component manually - let's say you have a <Modal /> component with a archived = exclude filter, and there are more than one of them in the query result. When you dismiss the modal, then we would optimistically mark the message as archived and that would trigger the same cycle of useInAppMessage() re-selecting and you end up getting a new (next) unarchived message which results in rendering another modal?

Yes happy to think through together as a team!

Reply:
`useInAppMessage` which the default components use makes a request with `page_size=1` so there will only be one message in the query info. There would have to be a refetch or a new message via ws to display a new message if I understand correctly, so I don't think it's actually as problematic for archiving.

Reply:
Ah yes that's true. This may only be a relevant issue when using `useInAppMessages` directly. 

Now i'm wondering if that creates a different but related issue.. If you have a `<Card />` component with an engagement filter set to uninteracted, then currently the moment you click on the card, it would disappear and that might feel surprising.

Maybe the only optimistic update we do that should trigger re-selection is archiving? We can discuss more on our call.

Reply:
As discussed, we're going to remove the optimistic filtering and let that be handled by the components.